Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -358,15 +358,19 @@ extension BrowserViewController: BrowserToolbarDelegate {
         }
     }
 
-    func browserToolbarDidLongPressBookmark(browserToolbar: BrowserToolbar) {
+    func browserToolbarDidLongPressBookmark(browserToolbar: BrowserToolbar, button: UIButton) {
     }
 
-    func browserToolbarDidPressShare(browserToolbar: BrowserToolbar) {
+    func browserToolbarDidPressShare(browserToolbar: BrowserToolbar, button: UIButton) {
         if let selected = tabManager.selectedTab {
             if let url = selected.displayURL {
-                var shareController = UIActivityViewController(activityItems: [selected.title ?? url.absoluteString!, url], applicationActivities: nil)
-                shareController.modalTransitionStyle = .CoverVertical
-                presentViewController(shareController, animated: true, completion: nil)
+                var activityViewController = UIActivityViewController(activityItems: [selected.title ?? url.absoluteString!, url], applicationActivities: nil)
+                if let popoverPresentationController = activityViewController.popoverPresentationController {
+                    popoverPresentationController.sourceView = browserToolbar
```

Comment:
Good catch. I think this can be simplified by just setting `sourceView` to `button` and not set `sourceView` at all. Double checking if there was a reason I chose the current code.